helm/reana/README.md

```diff
@@ -69,6 +69,7 @@ This Helm automatically prefixes all names using the release name to avoid colli
 | `components.reana_workflow_engine_snakemake.image`       | [REANA-Workflow-Engine-Snakemake image](https://hub.docker.com/r/reanahub/reana-workflow-engine-snakemake) to use | `docker.io/reanahub/reana-workflow-engine-snakemake:<chart-release-version>` |
 | `compute_backends`                                       | List of supported compute backends (kubernetes, htcondorcern, slurmcern)             | "kubernetes"                                    |
 | `dask.enabled`                                           | Enable support for running Dask workflows | false |
+| `dask.autoscaler_enabled`                                | Enable Dask autoscaler | true |
```

Comment:
Is one boolean variable enough? 

Looking at the Dask auto-scaler spec, some parameters are possible:

```yaml
apiVersion: kubernetes.dask.org/v1
kind: DaskAutoscaler
metadata:
  name: example
spec:
  cluster: "name of DaskCluster to autoscale"
  minimum: 0  # minimum number of workers to create
  maximum: 10 # maximum number of workers to create
```

Perhaps instead of using only one `autoscaler_enabled` Helm value we should use two for min and max, such as:

```yaml
dask.cluster_autoscale_workers_min: 1
dask.cluster_autoscale_workers_max: 4
```

And, if both min and max values would be equal to the same amount, then this would indicate that the auto-scaling would be disabled.

This would kind of "override"  the meaning of `cluster_default_number_of_workers` so we could get rid of that one...

We can think IRL.

Reply:
I am a bit susceptible about these new parameters. If they are equal for example, it not only means that auto-scaling is disabled but also users are not allowed to override the number of workers for their Dask cluster. 

I also think that we should be able to somehow restrict users to ask for unbounded number of workers (e.g 1000 workers with 1Mi memory` regardless of if the autoscaling is enabled or disabled. We might specify the max number of workers in another way by setting a minimum memory limit for one worker. This would implicitly indicate that `max_workers = max_dask_cluster_memory // min_worker_memory`. I think this approach is better since having a `min_worker_memory`. We should talk about how we approach this one IRL. 

And for the minimum number of workers, I do not see any case where we may want to force users to have at least X number of workers where X>1. So, maybe we can simply check if users request `<1 number of workers` and then reject such requests.

Reply:
OK, I see the point. We could keep the autoscale to be independent. However, allowing users to specify 1000 workers with low memory per worker would not be good, since users could generate 1000s of pods which would overload Kubernetes master and slow down everything... So we may want to keep memory independent. Let's muse IRL.